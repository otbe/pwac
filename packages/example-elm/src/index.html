<!DOCTYPE html>
<html>

<head>
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>
        PWA
    </title>

    <% if (htmlWebpackPlugin.files.favicon) { %>
        <link rel="shortcut icon" href="<%= htmlWebpackPlugin.files.favicon%>">
        <% } %>

</head>

<body>
    <div id="app"></div>
</body>

</html>