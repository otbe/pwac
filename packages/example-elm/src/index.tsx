import 'normalize.css';
import OfflinePluginRuntime from 'offline-plugin/runtime';
import { Main } from './Main.elm';

OfflinePluginRuntime.install({
  onInstalled: () => {
    console.log('SW Event:', 'onInstalled');
  },
  onUpdating: () => {
    console.log('SW Event:', 'onUpdating');
  },
  onUpdateReady: () => {
    console.log('SW Event:', 'onUpdateReady');
    OfflinePluginRuntime.applyUpdate();
  },
  onUpdated: () => {
    console.log('SW Event:', 'onUpdated');
    window.location.reload();
  },

  onUpdateFailed: () => {
    console.log('SW Event:', 'onUpdateFailed');
  }
});

console.log('foo'.includes('foo'));

Main.embed(document.getElementById('app'));
