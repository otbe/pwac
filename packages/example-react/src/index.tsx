import './style.css';
import 'normalize.css';
import React from 'react';
import { render } from 'react-dom';
import App from './App';
import OfflinePluginRuntime from 'offline-plugin/runtime';

OfflinePluginRuntime.install({
  onInstalled: () => {
    console.log('SW Event:', 'onInstalled');
  },
  onUpdating: () => {
    console.log('SW Event:', 'onUpdating');
  },
  onUpdateReady: () => {
    console.log('SW Event:', 'onUpdateReady');
    OfflinePluginRuntime.applyUpdate();
  },
  onUpdated: () => {
    console.log('SW Event:', 'onUpdated');
    window.location.reload();
  },

  onUpdateFailed: () => {
    console.log('SW Event:', 'onUpdateFailed');
  }
});

// babel + polyfills
console.log('foo'.includes('bar'));

render(<App />, document.getElementById('app'));
