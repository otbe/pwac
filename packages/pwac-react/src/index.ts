import * as Ajv from 'ajv';

export { prod } from './prod';
export { dev } from './dev';

export const validate = (config: any): boolean => {
  const ajv = new Ajv();
  ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-06.json'));
  ajv.validate(require('../config/schema.json'), config);
  return ajv.errors == null;
};
