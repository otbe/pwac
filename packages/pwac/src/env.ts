import * as webpack from 'webpack';

export const loadConfig = (
  production: boolean,
  projectConfig: any
): webpack.Configuration => {
  let projectEnv;

  try {
    projectEnv = require(`pwac-${projectConfig.env}`);
  } catch (e) {
    throw `Missing env ${projectConfig.env}`;
  }

  if (!projectEnv.validate(projectConfig)) {
    throw `invalid config for ${projectConfig.env} supplied`;
  }

  return projectEnv[production ? 'prod' : 'dev'](projectConfig);
};
