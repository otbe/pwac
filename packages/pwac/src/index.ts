import * as program from 'commander';
import { execSync, spawn } from 'npm-run';
import { resolve } from 'path';

const cwd = process.cwd();

program.version(require('../package.json').version);

program
  .command('watch')
  .option('--analyze-bundle')
  .action(async e => {
    spawn(
      'webpack-dev-server',
      [
        '--config',
        resolve(__dirname, 'webpack', 'dev.js'),
        e.analyzeBundle ? '--env.PWAC_ANALYZE_BUNDLE=true' : null
      ].filter(x => !!x),
      { cwd, stdio: 'inherit' }
    );
  });

program.command('build').action(async () => {
  try {
    execSync('tsc --noEmit', {
      cwd
    });
  } catch (e) {
    console.log(e.output[1].toString());
    process.exit(1);
    return;
  }

  spawn('webpack', ['--config', resolve(__dirname, 'webpack', 'prod.js')], {
    cwd,
    stdio: 'inherit'
  });
});

program
  .command('serve')
  .option('-p <port>')
  .action(async e =>
    spawn(
      'serve',
      ['-s', '-c 0', e.P ? `-p ${e.P}` : null, 'dist/'].filter(x => !!x),
      {
        cwd,
        stdio: 'inherit'
      }
    )
  );

program.on('command:*', function() {
  console.error(
    'Invalid command: %s\nSee --help for a list of available commands.',
    program.args.join(' ')
  );
  process.exit(1);
});

program.parse(process.argv);

if (program.args.length < 1) {
  program.help();
}
