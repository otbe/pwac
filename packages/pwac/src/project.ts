import { resolve } from 'path';
import { merge, memoize } from 'lodash';

const defaultConfigPath = process.env.PWAC_CONFIG || null;
const defaultPackageJsonPath = resolve(process.cwd(), './package.json');

export const getProjectConfig = memoize(
  (
    configPath: string | null = defaultConfigPath,
    packageJsonPath = defaultPackageJsonPath
  ): any =>
    merge(
      {},

      configPath ? require(configPath) : require(packageJsonPath).pwac || {}
    ),
  (arg0, arg1, arg2) =>
    `${arg0 || defaultConfigPath}${arg1 || defaultPackageJsonPath}${arg2}`
);

export const getWebpackModifierFn = (
  pwacConfigPath = resolve(process.cwd(), './pwac.config.js')
) => {
  let fn = (e: any) => e;

  try {
    fn = require(pwacConfigPath);
  } catch {}

  return fn;
};
