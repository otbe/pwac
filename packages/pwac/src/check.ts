import { getProjectConfig } from './project';
import { execSync } from 'npm-run';

export const check = () => {
  const config = getProjectConfig();
  const cwd = process.cwd();

  if (config.env === 'react') {
    try {
      execSync('tcm src/ -c -s');
    } catch (e) {
      console.log(e.output[1].toString());
      process.exit(1);
      return;
    }

    try {
      execSync('tsc --noEmit', {
        cwd
      });
    } catch (e) {
      console.log(e.output[1].toString());
      process.exit(1);
      return;
    }
  }
};
