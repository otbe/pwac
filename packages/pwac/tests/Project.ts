import { getProjectConfig, getWebpackModifierFn } from '../src/project';

describe('Project configuation', () => {
  it('should accept config file path via env', () => {
    expect(
      getProjectConfig(require.resolve(`./configs/complete.json`))
    ).toMatchSnapshot();
  });

  it('should return noop from missing pwac.config.js', () => {
    const o = {};
    expect(getWebpackModifierFn()(o)).toEqual(o);
  });

  it('should apply pwac.config.js', () => {
    const m = getWebpackModifierFn(require.resolve('./configs/pwac.config.js'))(
      {}
    );

    expect(m.test).toBeTruthy();
  });
});
