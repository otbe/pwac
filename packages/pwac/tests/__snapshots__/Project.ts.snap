// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project configuation should accept config file path via env 1`] = `
Object {
  "browsers": Array [
    ">2%",
  ],
  "env": "react",
  "images": Object {
    "sizes": Array [
      300,
      600,
      1200,
      2000,
    ],
  },
  "prerender": Object {
    "routes": Array [
      "/",
    ],
  },
}
`;
